mysql_ch_replicator/db_replicator.py

```diff
@@ -329,13 +354,11 @@ def perform_initial_replication_table(self, table_name):
 
             if not records:
                 break
+
             self.clickhouse_api.insert(table_name, records, table_structure=clickhouse_table_structure)
-            for record in records:
-                record_primary_key = [record[key_idx] for key_idx in primary_key_ids]
-                if max_primary_key is None:
-                    max_primary_key = record_primary_key
-                else:
-                    max_primary_key = max(max_primary_key, record_primary_key)
+
+            last_record = records[-1]
+            max_primary_key = [self.to_date_if_str(last_record[key_idx]) for key_idx in primary_key_ids]
```

Comment:
We should only attempt to convert for datetime or date types, not for arbitrary strings. Please do the following: 
1. Get MySQL field types for primary keys (add it after line 300):
   
       field_types_mysql = [field.field_type for field in mysql_table_structure.fields]
       primary_key_types_mysql = [field_types[key_idx] for key_idx in primary_key_ids]
       
2. Check if the type is datetime and only in this case try to convert it to dates.